File: B_PL_Codes/16-CódigoComFunções/4-ReordenarVetor/OrdenarMain.c
#include <stdio.h>
#include "lib.h"
#include <stdlib.h>
int main()
{

    int v[] = {10, 8, 12, 3, 5};
    // Determinar o tamanho do vetor
    int n = sizeof(v) / sizeof(v[0]);
    int metodo;
    do
    {
        printf("\nEscolha um método de ordenação do vetor:"
               "\n(1) - Método de permuta direta"
               "\n(2) Método do bolha\n");
        scanf("%d", &metodo);

        switch (metodo)
        {
        case 1:
            system("clear");
            printf("Vetor original:");
            // Imprimir vetor original
            ImprimirVetor(v, n);
            // Organizar vetor
            OrdenarVetorMet01(v, n);
            // Imprimir vetor
            printf("Vetor ordenado utilizando o método de permuta:");
            ImprimirVetor(v, 5);
            break;

        case 2:
            system("clear");
            printf("Vetor original:");
            ImprimirVetor(v, n);
            OrdenarVetorMet02(v, n);
            printf("Vetor ordenado utilizando o método do bolha:");
            ImprimirVetor(v, 5);
            break;
        default:
            printf("Método fora da lista de opções!");
            break;
        }

        printf("\nDigite (0) para encerrar ou qualquer tecla para reiniciar.");
        scanf("%d", &metodo);
        system("clear");
    } while (metodo != 0);
    printf("Código encerrado!");
}

File: B_PL_Codes/16-CódigoComFunções/4-ReordenarVetor/InJava/OrdenarMain.java

import java.util.Scanner;

public class OrdenarMain {

    public static void main(String[] args) {

        int[] v = { 10, 8, 12, 3, 5 }; //
        int metodo;

        Scanner dados = new Scanner(System.in);

        do {

            RotinasOrdenar.cls(); // Método criado - equivalente ao system("clear") do <stdlib.h> no C;
            System.out.println(
                    "\nEscolha um método de ordenação do vetor:" +
                            "\n(1) - Método de permuta direta" +
                            "\n(2) - Método do bolha" +
                            "\nSua escolha: ");
            metodo = dados.nextInt();
            switch (metodo) {
                case 1:
                    System.out.printf("Vetor original:");
                    // Imprimir vetor original
                    RotinasOrdenar.ImprimirVetor(v);
                    // Organizar vetor
                    RotinasOrdenar.OrdenarVetorMet01(v);
                    // Imprimir vetor
                    System.out.printf("Vetor ordenado utilizando o método de permuta:");
                    RotinasOrdenar.ImprimirVetor(v);
                    break;

                case 2:
                    System.out.printf("Vetor original:");
                    RotinasOrdenar.ImprimirVetor(v);
                    RotinasOrdenar.OrdenarVetorMet02(v);
                    System.out.printf("Vetor ordenado utilizando o método do bolha:");
                    RotinasOrdenar.ImprimirVetor(v);
                    break;

                default:
                    System.out.printf("Método fora da lista de opções!");
                    break;
            }

            System.out.printf("\nDigite (0) para encerrar ou qualquer tecla para reiniciar.");
            metodo = dados.nextInt();
        } while (metodo != 0);

        System.out.printf("Código encerrado!");
        dados.close();
    }
}

// Determinar o tamanho do vetor
/*
 * Em Java, a convenção é colocar os colchetes junto ao tipo (int[])
 * em vez de junto ao nome da variável, como é comum em C.
 * Por isso:
 * int[] v= {1,2,3};
 * 
 * 
 */